packages/core/src/lib/code-snippet.svelte

```diff
@@ -0,0 +1,230 @@
+<!--
+@component
+  
+For rendering readonly blocks of code. Uses `prism` for syntax highlighting:
+
+https://prismjs.com
+
+```svelte
+<CodeSnippet language="json" code="{ myThing: true }">
+    This text captions the code snippet.
+</CodeSnippet>
+```
+-->
+<svelte:options immutable />
+
+<script
+  context="module"
+  lang="ts"
+>
+import Prism from 'prismjs';
```

Comment:
Is this statically importing all of Prism? That seems concerning

Reply:
I actually think I can just import the function we need according to a doc I was looking at but forgot to try it out. Will give it a go tomorrow.

Reply:
Appears to be working, we just need to import the `highlightElements` function and the `plugins` Record.